apps/web-giddh/src/app/subscription/subscription-list/subscription-list.component.ts

```diff
@@ -0,0 +1,755 @@
+            }
+            if (searchedText === null || searchedText === "") {
+                this.showClearFilter = false;
+                this.showStatus = false;
+            }
+        });
+
+        this.subscriptionListForm?.controls['duration'].valueChanges.pipe(
+            debounceTime(700),
+            distinctUntilChanged(),
+            takeUntil(this.destroyed$),
+        ).subscribe(searchedText => {
+            if (searchedText !== null && searchedText !== undefined) {
+                this.showClearFilter = true;
+                this.getAllSubscriptions(true);
+            }
+            if (searchedText === null || searchedText === "") {
+                this.showClearFilter = false;
+                this.showMonthlyYearly = false;
+            }
+        });
+
+        this.componentStore.isUpdateCompanySuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.getAllSubscriptions(null);
+            }
+        });
+
+        this.archiveCompanySuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                let text = this.localeData?.company_message;
+                text = text?.replace("[TYPE]", response?.archiveStatus === 'USER_ARCHIVED' ? this.commonLocaleData?.app_unarchive : this.commonLocaleData?.app_archive);
+                this.toasterService.showSnackBar('success', text);
+                this.getAllSubscriptions(null);
+            }
+        });
+
+    }
+
+    /**
+     * This will use for init subscription form
+     *
+     * @memberof SubscriptionComponent
+     */
+    public initForm(): void {
+        this.subscriptionListForm = this.formBuilder.group({
+            companyName: null,
+            billingAccountName: null,
+            subscriberName: null,
+            countryName: null,
+            planName: null,
+            status: null,
+            duration: null
+        });
+    }
+
+    /**
+   * Returns the search field text
+   *
+   * @param {*} title
+   * @returns {string}
+   * @memberof SubscriptionComponent
+   */
+    public getSearchFieldText(title: any): string {
+        let searchField = this.localeData?.search_field;
+        searchField = searchField?.replace("[FIELD]", title);
+        return searchField;
+    }
+
+    /**
+     * Handles clicks outside the specified element for filtering in the SubscriptionComponent.
+     *
+     * @param event - The event triggered by the click.
+     * @param element - The element outside of which the click occurred.
+     * @param searchedFieldName - The name of the field being searched for.
+     * @memberof SubscriptionComponent
+     */
+    public handleClickOutside(event: any, element: any, searchedFieldName: string): void {
+        if (searchedFieldName === 'companyName') {
+            if (this.subscriptionListForm?.controls['companyName'].value !== null && this.subscriptionListForm?.controls['companyName'].value !== '') {
```

Comment:
Confirm with @ravindersinghdang is this possible to write in single if instead of nested if

if (searchedFieldName === 'companyName' && (this.subscriptionListForm?.controls['companyName'].value !== null && this.subscriptionListForm?.controls['companyName'].value !== '')) {